src/fetcher.ts

```diff
@@ -193,6 +194,8 @@ export interface AutoInitOptions extends RequestInit{
   body?: string
   headers: HeadersInit
   credentials?: 'include' | 'omit'
+  // for local storage support
+  inBrowserStorage?: boolean
```

Comment:
Why do users need to add this? If we default to true, both http(s) and localStorage would work, so from a developer experience point of view that's better. Maybe the reason is that using this without actually using localStorage will always make an extra call with that if that checks for the `lp://` prefix, but I think the performance impact is negligible.

Reply:
I suppose one reason to make this opt-in would be that many applications would not need this functionality. I've only recently heard about tree shaking, so I'm not sure what it would involve here, but in principle we might want a developer to only import this fetcher if they need it?

Reply:
Yes, we could change to always-on.  It would add a single if statement to every fetch.  Not sure if that matters.

Reply:
I think it's better to keep a simple API, so the less features that are configurable, the better. That is, if there is no reason to make it configurable of course.

What @josephguillaume mentions about tree-shaking would be a decent reason to make it opt-in, although at this point we wouldn't benefit a lot from tree-shaking the localStorage functionality because it's very small (but it could grow). But if we want to enable tree-shaking, it cannot be configured with an option. It should be implemented with some hooks or pluginable interface. Basically, applying the [inversion of control](https://en.wikipedia.org/wiki/Inversion_of_control) principle. Which in essence, can already be done with the `fetch` argument :). So maybe a better API, if we want to enable that, would be:

```js
$rdf.fetcher(kb, { fetch: localStorageFetch() });
```

Reply:
We definitely do not want to use the fetch param.  Say I have a Dropbox fetch or a file:// fetch using solid-rest or even an inrupt-client fetch ... the app can define those with the fetch flag. 
 The current PR leaves those in place and adds the localStorage part on top of them.

Reply:
Also, the way the PR does it supports the user over-riding the browser fetch.  They can set fetch in the call to fetcher to their own inBrowserStorage and leave inBrowserStorage false,

Reply:
Given that the code doesn't to seem to import much that's new, I agree that it's probably a minimal increase to the bundle size and it doesn't matter in this case. I just think this is something we should be thinking about. I've recently been bothered by the size of jsonld when all I wanted was two functions. (I haven't yet worked out of that's fixable)

Following Noel's suggestion, if bundle size were important, then I would suggest a function e.g. createInBrowserFetch that took the user fetch as an argument and returned an augmented fetch for use in fetcher.
Alternatively, a wrapper around this constructor specifically for in-browser capabilities. 
A developer then has the choice on whether this code is in the bundle.